data/startup.sh

```diff
@@ -141,6 +141,84 @@ fetch_builtin_ctx() {
   echo "Done fetching builtin build context"
 }
 
+# need to forcefully stop journald service to unmount stateful partition.
+stop_journald_service() {
+  mkdir -p /etc/systemd/system/systemd-journald.service.d
+  cat > /etc/systemd/system/systemd-journald.service.d/override.conf<<EOF
+[Service]
+Restart=no
+EOF
+
+  /usr/bin/systemctl daemon-reload
+  /usr/bin/systemctl stop systemd-journald.socket
+  /usr/bin/systemctl stop systemd-journald-dev-log.socket
+  /usr/bin/systemctl stop systemd-journald-audit.socket
+  /usr/bin/systemctl stop syslog.socket
+  /usr/bin/systemctl stop systemd-journald.service
+}
+
+# this unit runs at shutdown time after everything but /tmp is unmounted
+create_run_after_unmount_unit(){
+  mkdir /tmp/last
+  mount -t tmpfs tmpfslast /tmp/last
```

Comment:
nit: it's more typical to have the first argument to mount be "none" or "tmpfs":

mount -t tmpfs tmpfs "${tmpdir}" or mount -t tmpfs none "${tmpdir}"